common/rdr/AESInStream.cxx

```diff
@@ -0,0 +1,106 @@
+/* Copyright (C) 2022 Dinglan Peng
+ * 
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <rdr/AESInStream.h>
+#include <rdr/Exception.h>
+
+#ifdef HAVE_GNUTLS
+using namespace rdr;
+
+const size_t MaxMessageSize = 65536;
+
+AESInStream::AESInStream(InStream* _in, const U8* key, int _keySize)
+  : keySize(_keySize), in(_in), msgLength(0), msgOffset(0), counter{0}
+{
+  msg = new U8[MaxMessageSize];
```

Comment:
Sure. It's merged now as 0ae1557ed9a7f8b46b6d7e2feaf2fad4cc2c3b3d .

Your PR here exposed some more issues in the existing code. Do you want me to merge any of these:

 * 6bd09e8c38f5dbb3f9b473ce1725a4d15ea8d52e
 * f1bfbd3dd74169ef6f4d83f0b3513475d0fcdac2
 * 66da1b2adf7ca9f2e177830ec87377ed248cd105
 * 83031fc1d466d78481c069ae4a075de182a87694

Otherwise, I'll just clean up things once everything here is merged.